index.js

```diff
@@ -15,20 +18,14 @@ const last = (predicate, resultSelector) => source => (start, slink) => {
       talkback = d;
       slink(t, d);
     } else if (t === 1) {
-      if (predicate) {
-        if (predicate(d)) {
-          lastVal = d;
-          matched = true;
-        }
-      } else {
+      if (predicate(d)) {
```

Comment:
by providing a default value for `predicate` we can use a single if branch here instead of 2